lib/xmlrpc/parser.rb

```diff
@@ -117,10 +117,14 @@ def self.struct(hash)
           obj = mod.allocate
 
           hash.delete "___class___"
```

Comment:
We should not delete `___class___` when we don't create an object.

How about the following?

```diff
diff --git a/lib/xmlrpc/parser.rb b/lib/xmlrpc/parser.rb
index a58da33..9ffbcd0 100644
--- a/lib/xmlrpc/parser.rb
+++ b/lib/xmlrpc/parser.rb
@@ -113,6 +113,7 @@ module XMLRPC # :nodoc:
         begin
           mod = Module
           klass.split("::").each {|const| mod = mod.const_get(const.strip)}
+          return hash unless mod.included_modules.include?(Marshallable)
 
           obj = mod.allocate
 
```